From 844aba7d79e7a68cdf5876cc1659eab7a5e16b5b Mon Sep 17 00:00:00 2001
From: Hannes Schmelzer <hannes.schmelzer@br-automation.com>
Date: Tue, 26 Sep 2017 13:14:45 +0200
Subject: [PATCH] disable MOCK

Signed-off-by: Hannes Schmelzer <oe5hpm@oevsv.at>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b369a6d..faa07a8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -164,7 +164,7 @@ add_subdirectory(src)
 
 find_package(Git)
 
-set(ENABLE_GOOGLEMOCK TRUE)
+set(ENABLE_GOOGLEMOCK FALSE)
 #disable unit tests on gcc4.6 (problems with googlemock 1.7)
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
     # require at least gcc 4.8
-- 
2.7.4

